internal/walk/walklinks.go

```diff
@@ -20,8 +21,22 @@ func pathRewriter(src string, dest string, fn fs.WalkDirFunc) fs.WalkDirFunc {
 	}
 }
 
+func makeRegexp(str string) (re *regexp.Regexp, err error) {
```

Comment:
The logic here isn't going to work right in several cases:

1. If str contains regex special characters like `.` .  Example which will actually occur is "repodata.old" - this will wrongly match any character before "old", like "repodataxold".  There'd need to be a QuoteMeta here probably.
2. If str contains a character class match like `[a-z]` - this *would* actually work right now, but then if you fix (1) by QuoteMeta, it'll become broken.
3. If str contains `***`.
4. If str contains any escaped characters, e.g. `\*` should disable special handling of `*`

While I don't think it's necessary to fully support all of these, I also don't think we should just have undefined behavior without so much as a comment acknowledging that the code isn't complete.

From the above issues (1) is the only one I'm sure will actually come into play in real usage, so we should fix that one at least.  For (2-4) I'm not sure they're important, but could we at least detect those cases and return some kind of "not supported" error rather than silently doing the wrong thing?

I also strongly recommend adding a test for makeRegexp specifically, it's a pure function which takes a string as output and returns a regexp or error, it'll be much more efficient to test some of the edge cases by directly calling it rather than trying to cover everything through high-level tests.